apps/publish/content/common.py

```diff
@@ -914,6 +918,31 @@ def _inherit_publish_schedule(self, original, updates, associated_item):
                 associated_item[PUBLISH_SCHEDULE] = publish_schedule
                 associated_item[SCHEDULE_SETTINGS] = schedule_settings
 
+    def _update_picture_metadata(self, updates, original, updated):
```

Comment:
_This is just a personal preference._
To avoid nested if statements, and along the idea of `fail fast / return early`, this method could be re-written as:

```python
def _update_picture_metadata(self, updates, original, updated):
    renditions = updated.get("renditions") or {}
    mapping = app.config.get("PHOTO_METADATA_MAPPING")
    if not mapping or not renditions:
        return

    try:
        media_id = renditions["original"]["media"]
    except (KeyError, TypeError):
        return

    if not media_id:
        return

    picture = app.media.get(media_id)
    binary = picture.read()
    metadata = get_metadata_from_item(updated, mapping)
    updated_binary = write_metadata(binary, metadata)

    if updated_binary == binary:
        return

    updated_media_id = app.media.put(
        updated_binary, content_type=picture.content_type, filename=picture.filename
    )
    updates.setdefault("renditions", deepcopy(renditions))["original"].update(
        {
            "media": updated_media_id,
            "href": app.media.url_for_media(updated_media_id, picture.content_type),
        }
    )
    updated["renditions"] = updates["renditions"]
```